File: Ruby/CommandARGVAnalyze.rb

module CommandAnalyzeKeyType
  Default = "Default"
  String = "String"
end

class CommandARGVAnalyze
  attr_accessor :mark
  attr_accessor :keyType
  def self.default(params = {})
    if !params.has_key?(:keyType)
      params[:keyType] = CommandAnalyzeKeyType::Default
    end
    return CommandARGVAnalyze.new(params)
  end
  def initialize(params = {})
    @mark = params.has_key?(:mark) ? params[:mark] : ":"
    @keyType = params.has_key?(:keyType) ? params[:keyType] : CommandAnalyzeKeyType::String
  end
  # 冒号分隔，仅返回符合条件的
  def tiskMark(params)
    a, b = tiskMarkFull(params)
    return a
  end
  # 冒号分隔，返回两个变量，第一个符合条件的hash值，第二个不符合条件的源数据
  def tiskMarkFull(params)
    dic = {}
    other = []
    params.each { |obj|
      if obj.class != String || !obj.include?(@mark)
        other << obj
        next
      end
      markIndex = obj.index(@mark)
      if markIndex == 0 || markIndex == obj.length
        other << obj
        next
      end
      key = obj[0...markIndex]
      value = obj[markIndex + 1 ... obj.length]
      if @keyType == CommandAnalyzeKeyType::String
        dic[key] = value
      else
        dic[:"#{key}"] = value
      end
    }
    return dic, other
  end
end

File: Python/Analyze/commandARGVAnalyze.py

import sys

def tiskMark(params, mark = ":"):
    return tiskMarkFull(params, mark)[0]

def tiskMarkFull(params, mark = ":"):
    dic = {}
    other = []
    for obj in params:
        index = obj.find(mark)
        if index == -1 or index == 0 or index == len(obj):
            other.append(obj)
            continue
        key = obj[0 : index]
        value = obj[index + 1 : len(obj)]
        dic[key] = value
    return (dic, other)
